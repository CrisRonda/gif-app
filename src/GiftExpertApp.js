import React, { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";
import PropTypes from "prop-types";

const GiftExpertApp = ({ initialState }) => {
  const [params, setParams] = useState(initialState);
  return (
    <>
      <h2>GiftExpertApp</h2>
      <AddCategory setParams={setParams} />
      <ol>
        <GifGrid category={params.category} limit={params.limit} />
      </ol>
    </>
  );
};
GiftExpertApp.propTypes = {
  initialState: PropTypes.shape(),
};

GiftExpertApp.defaultProps = {
  initialState: {
    category: "Rainbow",
    limit: 10,
  },
};
export default GiftExpertApp;
