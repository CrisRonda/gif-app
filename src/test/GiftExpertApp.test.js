import { shallow } from "enzyme";
import GiftExpertApp from "../GiftExpertApp";

describe("Test on GiftExpertApp", () => {
  test("should render compontent", () => {
    let wrapper = shallow(<GiftExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should render items on gifgrid", () => {
    const state = { category: "Goku", limit: 5 };
    let wrapper = shallow(<GiftExpertApp initialState={state} />);
    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(1);
  });
});
