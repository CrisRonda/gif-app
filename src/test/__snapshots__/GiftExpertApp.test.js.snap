// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GiftExpertApp should render compontent 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setParams={[Function]}
  />
  <ol>
    <GifGrid
      category="Rainbow"
      limit={10}
    />
  </ol>
</Fragment>
`;
